lambdas/data-migration2/src/storeErrors.ts

```diff
@@ -0,0 +1,58 @@
+import Logger from '@cumulus/logger';
+
+const fs = require('fs');
+const JSONStream = require('JSONStream');
+// const { finished } = require('stream');
+// const { promisify } = require('util');
+const { s3 } = require('@cumulus/aws-client/services');
+
+const logger = new Logger({ sender: '@cumulus/data-migration/storeErrors' });
+
+/**
+ * Helper to create error file write stream
+ * @param {string} migrationName         - Name of migration
+ * @param {string | undefined} timestamp - Timestamp for unit testing
+ * @returns {Object}                     - Object containing error write streams and file path
+ */
+export const createErrorFileWriteStream = (migrationName: string, timestamp?: string) => {
+  const dateString = timestamp || new Date().toISOString();
+  const filepath = `${migrationName}ErrorLog-${dateString}.json`;
+  const errorFileWriteStream = fs.createWriteStream(filepath);
+  const jsonWriteStream = JSONStream.stringify('{"errors": [\n', '\n,', '\n]}\n');
+  jsonWriteStream.pipe(errorFileWriteStream);
+
+  return { jsonWriteStream, errorFileWriteStream, filepath };
+};
+
+/**
+ * Store migration errors JSON file on S3.
+ *
+ * @param {Object} params
+ * @param {string} params.bucket                - Name of S3 bucket where file will be uploaded
+ * @param {string} params.filepath              - Write Stream file path
+ * @param {string} params.migrationName         - Name of migration
+ * @param {string} params.stackName             - User stack name/prefix
+ * @param {string | undefined} params.timestamp - Timestamp for unit testing
+ * @returns {Promise<void>}
+ */
+export const storeErrors = async (params: {
+  bucket: string,
+  filepath: string,
+  migrationName: string,
+  stackName: string,
+  timestamp?: string,
+}) => {
+  const { bucket, filepath, migrationName, stackName, timestamp } = params;
+  const fileKey = `data-migration2-${migrationName}-errors`;
+  const dateString = timestamp || new Date().toISOString();
+  const key = `${stackName}/${fileKey}-${dateString}.json`;
+
+  await s3().putObject({
+    Bucket: bucket,
+    Key: key,
+    Body: fs.createReadStream(filepath),
+  }).promise();
+
+  logger.info(`Stored error log file with key ${key} to bucket ${bucket}.`);
```

Comment:
```suggestion
  logger.info(`Stored error log file on S3 at s3://${bucket}/${key}`);
```

This would make it easier for someone to copy/paste the S3 location into their terminal and use it with a CLI command

Reply:
Ooh that is smart!